vertexai/genai/tokenizer/corpora_test.go

```diff
@@ -0,0 +1,305 @@
+		{Pattern: regexp.MustCompile(".*-SJIS$"), Encoding: japanese.ShiftJIS},
+		{Pattern: regexp.MustCompile(".*-GB2312$"), Encoding: simplifiedchinese.HZGB2312},
+		{Pattern: regexp.MustCompile(".*-Latin2$"), Encoding: charmap.ISO8859_2},
+		{Pattern: regexp.MustCompile(".*-Greek$"), Encoding: charmap.ISO8859_7},
+		{Pattern: regexp.MustCompile(".*-UTF8$"), Encoding: encoding.Nop}, // No transformation needed
+		{Pattern: regexp.MustCompile("Amahuaca"), Encoding: charmap.ISO8859_1},
+		{Pattern: regexp.MustCompile("Turkish_Turkce-Turkish"), Encoding: charmap.ISO8859_9},
+		{Pattern: regexp.MustCompile("Lithuanian_Lietuviskai-Baltic"), Encoding: charmap.ISO8859_4},
+		{Pattern: regexp.MustCompile("Japanese_Nihongo-EUC"), Encoding: japanese.EUCJP},
+		{Pattern: regexp.MustCompile(`Abkhaz\-Cyrillic\+Abkh`), Encoding: charmap.Windows1251},
+	}
+
+	skip := map[string]bool{
+		"Burmese_Myanmar-UTF8":           true,
+		"Japanese_Nihongo-JIS":           true,
+		"Chinese_Mandarin-HZ":            true,
+		"Chinese_Mandarin-UTF8":          true,
+		"Gujarati-UTF8":                  true,
+		"Hungarian_Magyar-Unicode":       true,
+		"Lao-UTF8":                       true,
+		"Magahi-UTF8":                    true,
+		"Marathi-UTF8":                   true,
+		"Tamil-UTF8":                     true,
+		"Vietnamese-VPS":                 true,
+		"Vietnamese-VIQR":                true,
+		"Vietnamese-TCVN":                true,
+		"Magahi-Agra":                    true,
+		"Bhojpuri-Agra":                  true,
+		"Esperanto-T61":                  true,
+		"Burmese_Myanmar-WinResearcher":  true,
+		"Armenian-DallakHelv":            true,
+		"Tigrinya_Tigrigna-VG2Main":      true,
+		"Amharic-Afenegus6..60375":       true,
+		"Navaho_Dine-Navajo-Navaho-font": true,
+		"Azeri_Azerbaijani_Cyrillic-Az.Times.Cyr.Normal0117": true,
+		"Azeri_Azerbaijani_Latin-Az.Times.Lat0117":           true,
+		"Czech-Latin2-err":     true,
+		"Russian_Russky-UTF8~": true,
+	}
+
+	return &udhrCorpus{
+		Encodings: encodings,
+		Skip:      skip,
+	}
+}
+
+// getEncoding returns the encoding for a given filename based on patterns
+func (ucr *udhrCorpus) getEncoding(filename string) (encoding.Encoding, bool) {
+	for _, pattern := range ucr.Encodings {
+		if pattern.Pattern.MatchString(filename) {
+			return pattern.Encoding, true
+		}
+	}
+	return nil, false
+}
+
+// shouldSkip checks if the file should be skipped
+func (ucr *udhrCorpus) shouldSkip(filename string) bool {
+	return ucr.Skip[filename]
+}
+
+// decodeBytes converts a byte slice from a specified encoding to a UTF-8 string
+func decodeBytes(enc encoding.Encoding, data []byte) (string, error) {
+	if enc == encoding.Nop {
+		return string(data), nil
+	}
+
+	if enc == nil {
+		return "", fmt.Errorf("unsupported encoding or custom handling required")
+	}
+
+	decoder := enc.NewDecoder()
+	reader := transform.NewReader(strings.NewReader(string(data)), decoder)
```

Comment:
Could you use https://pkg.go.dev/golang.org/x/text/transform#Bytes here instead of a reader?

Reply:
Done

Reply:
Done